modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurator/ClientExtensionProjectConfigurator.java

```diff
@@ -1161,6 +1220,43 @@ private void _validateClientExtension(
 		}
 	}
 
+	private void _validateGlobalJSScriptElementAttributeName(String attribute) {
+		if (attribute.equals("src")) {
+			throw new GradleException(
+				"Use the 'url' field instead of the script attribute 'src'");
+		}
+	}
+
+	private void _validateGlobalJSScriptElementAttributes(
+		ClientExtension clientExtension) {
+
+		Map<String, Object> typeSettings = clientExtension.typeSettings;
+
+		List<Object> scriptElementAttributes =
+			(ArrayList<Object>)typeSettings.get("scriptElementAttributes");
+
+		if (scriptElementAttributes == null) {
+			return;
+		}
+
+		for (Object attribute : scriptElementAttributes) {
+			if (attribute instanceof String) {
+				_validateGlobalJSScriptElementAttributeName((String)attribute);
+			}
+			else if (attribute instanceof LinkedHashMap) {
```

Comment:
From the [YAML docs](https://yaml.org/type/bool.html), it seems there are no such thing as attributes with no values. We should probably follow that, it also makes the code a little bit easier. If the user adds the attribute with no value:

```
scriptElementAttributes
    - defer
```

We would show a message similar to `A value for the attribute 'defer' must be defined` on the console and UI.

What do you say?

Reply:
Well, I think we could make it even a little easier. scriptElementAttributes could be a map rather than an array, so it should just look like this:
```
scriptElementAttributes:
    defer: true
```

If we do it like this, I wonder what the error would be if instead you added:
```
scriptElementAttributes:
    defer:
```
or
```
scriptElementAttributes:
    defer
```

Reply:
@ethib137  In
```
scriptElementAttributes:
    defer:
```
defer is mapped to null

in
```
scriptElementAttributes:
    defer
```

the object mapper throws an error.

Should we care about the null?

`scriptElementAttributes=data-whatever=\"oi\"\ndefer=\"null\"`